cmake/modules/FindGnuLd.cmake

```
@@ -43,10 +43,10 @@ else()
 
   find_program(GNULD_LINKER ${CROSS_COMPILE}ld.bfd ${LD_SEARCH_PATH})
   if(GNULD_LINKER)
-    set(GNULD_LINKER_IS_BFD TRUE)
+    set(GNULD_LINKER_IS_BFD ON CACHE BOOL "Is linker ld.bfd compatible")
```

please add `FORCE`.

If this code ever executes a second time, then it means that someone purposely has cleared the `GNULD_LINKER` setting, and therefor we should ensure to update the corresponding `GNULD_LINKER_IS_BFD` to the new lookup we perform.

```suggestion
    set(GNULD_LINKER_IS_BFD ON CACHE BOOL "Is linker ld.bfd compatible" FORCE)
```

---

I added the code to return early if GNULD_LINKER_IS_BFD is already set. So no need to FORCE a change.

---

@dcpleung your "return early" is currently when GNULD_LINKER exists. @tejlmand describes a case where someone deleted it.

---

Added back.